### PR TITLE
First pass at an update to the process document.

### PR DESCRIPTION
**Pull request by waynebeaton**
This includes the definition of a new Progress Review to address Bug 496321 and (at least in part) Bug 534828. I've noted other changes in the CHANGELOG file.

This is intended as a first draft of the changes for consideration and review.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# ChangeLog
```
Comment (msohn) [inline CHANGELOG:1]: name this file CHANGELOG.md to render it as markdown on Github or also use asciidoc for the changelog ?

File: eclipse_development_process.adoc [hunk 2]
```diff
@@ -406,23 +409,21 @@ _(Most of this section is borrowed and paraphrased from the excellent http://www
 
 Releases are, by definition, anything that is distributed outside of the committers of a project. If users are being directed to download a build, then that build has been released (modulo the exceptions below). All projects and committers must obey the Eclipse Foundation requirements on approving any release.
 
+A project can make official releases for one calendar year following a successful link:#6_3_5_Progress_Review[progress review] or link:#6_3_3_Release_Review[release review]. The project leadership chain may--at its discretion--require that the project engage in additional reviews (e.g. a progress or release review) prior to issuing a release.
```
Comment (msohn) [inline eclipse_development_process.adoc:412]: - do projects still need to send IP logs for approval for each release which is not a bug fix release ?
- do we have any standards which should/must be followed for each release, not regarding if there is a release review for the release or not, e.g. each release must have release notes

File: CHANGELOG (hunk 1 above)
Comment (guw) [reply CHANGELOG:1]: I think this is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). Thus, CHANGELOG.md would be the preferred file name.

File: CHANGELOG [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## [Unreleased]
+
+### Added
+- (Bug 496321) The notion of a Progress Review added to augment (and in some cases, replace) Release Reviews.
```
Comment (guw) [inline CHANGELOG:6]: This is just a minor issue. I'd prefer to have the bug reference *after* the text. It improves readability. I think you should also turn the bug reference into a link. This is supported because it's Markdown.

File: CHANGELOG [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- Content in the "Releases" has been reorganized.
```
Comment (guw) [inline CHANGELOG:7]: This seems to be a duplicate of what's already in the Changes section. Should be removed here.

File: CHANGELOG [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+-- The requirement that a project engage in a Release Review prior to a release has been changed to a requirement that project can only release within a year of successfully engaging in either a Progress Review or a Release Review.
```
Comment (guw) [inline CHANGELOG:8]: I'd reword this a bit for readability purposes. Also, I don't think it needs to be a sub-bullet.

_Proposal:_
- The requirement that a project engages in a Release Review prior to a release has been replaced. Projects can now release as often as they want within a year after they successfully engaged in either a Progress Review or a Release Review.

File: CHANGELOG [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+### Changes
+- Converted document source format to Asciidoctor.
+- Removed the unnecessary statement that "Major releases continue to go through progress reviews." from the description of the Mature Phase.
```
Comment (guw) [inline CHANGELOG:12]: This should go into a "Removed" section.

File: CHANGELOG [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+- Release Reviews have been recast as a special type of Progress Review.
```
Comment (guw) [inline CHANGELOG:13]: I find this confusing. If Release Reviews are a special type of Progress Review, why do we treat them as separate things above?

File: eclipse_development_process.adoc [hunk 8]
```diff
@@ -360,18 +361,20 @@ The graduation review confirms that the project is/has:
...
 
 [#6_3_4_Promotion_Review]
 ==== 6.3.4 [Reserved]
 
-[#6_3_5_Continuation_Review]
-==== 6.3.5 [Reserved]
+[#6_3_5_Progress_Review]
+==== 6.3.5 Progress Review
+
+The purposes of a progress review are: to summarize the accomplishments of the project, to verify that the IP Policy has been followed and all approvals have been received, to highlight any remaining quality and/or architectural issues, and to verify that the project is continuing to operate according to the principles and purposes of Eclipse.
```
Comment (guw) [inline eclipse_development_process.adoc:377]: I wonder if a Progress Review shouldn't require that _"all approvals have been received"_. This seems more like a condition for a Release Review.

* A Release Review is a Progress Review tight to a release.
* In addition to a Progress Review, a Release Review requires that all IP approvals have been received.

File: eclipse_development_process.adoc (hunk 2 above)
Comment (guw) [reply eclipse_development_process.adoc:412]: @msohn 

_"do projects still need to send IP logs for approval for each release"_ -> No

This will be done only during a Release or Progress review. An IP log is not (never was) tight to a release. It was verified as part of the Release Review but technically an IP log represents the whole project.

_"do we have any standards which should/must be followed for each release"_ -> Yes

Yes. However, maybe we need some additional wording here. 

@waynebeaton I think technically the documentation requirement was captured by the Release Review definition in combination with _"All projects and committers must obey the Eclipse Foundation requirements on approving any release."_. Should there be some wording here that releases still require some documentation (I'm thinking Changelog, release record) and PMC approval?

File: eclipse_development_process.adoc [hunk 9]
```diff
@@ -406,23 +409,21 @@ _(Most of this section is borrowed and paraphrased from the excellent http://www
...
 
 _(Exception 2: milestone and release candidate builds)_ Projects are encouraged to use an agile development process including regular milestones (for example, six week milestones). Milestones and release candidates are "almost releases" intended for adoption and testing by early adopters. Projects are allowed to have links on the project website, blogs, wikis, etc. to encourage these outside-the-committer-circle early adopters to download and test the milestones and release candidates, but such communications must include caveats explaining that these are not official releases.
 
+_(Exception 3: bug fix releases with no new features)_ Bug fix releases (x.y.z, e.g., 2.3.1) with no significant changes or additions over the base release (e.g., 2.3) are allowed to be released without an additional review.
```
Comment (guw) [inline eclipse_development_process.adoc:418]: Maybe clarify that _"without an additional review"_ extends beyond the one calendar year period, i.e. a project can do a bug fix release even 13, 14 or 27+x months after its last progress/release review?

File: history.adoc [hunk 10]
```diff
@@ -1,4 +1,6 @@
 [#8_2_History]
 ==== 8.2 History
 
-Changes made in this document:
\ No newline at end of file
+Changes made in this document:
+
+include::CHANGELOG[leveloffset=+4]
```
Comment (guw) [inline history.adoc:6]: nice

File: CHANGELOG (hunk 1 above)
Comment (waynebeaton) [reply CHANGELOG:1]: +1

File: eclipse_development_process.adoc (hunk 2 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:412]: I generally try to avoid putting implementation details in the process document. IP Log reviews are how we meet the requirements of the IP Policy, which doesn't mention IP Logs specifically. That is, the IP Policy states "The Release Review is used as a final review to ensure that all necessary rights to the new Content have, in fact, been acquired."; the way that we do this is via an IP Log review.

File: CHANGELOG (hunk 3 above)
Comment (waynebeaton) [reply CHANGELOG:6]: Sure. We can do that.

**Comment by waynebeaton** [comment]
I don't see anything in the comments that's too controversial. Further, this isn't intended to be 100% complete. Can we accept this pull request so that others can more directly contribute patches?

**Comment by msohn** [comment]
looks good to me

File: CHANGELOG (hunk 4 above)
Comment (waynebeaton) [reply CHANGELOG:7]: Agreed. I've reorganized the bullets.

File: CHANGELOG (hunk 5 above)
Comment (waynebeaton) [reply CHANGELOG:8]: Agreed.

File: CHANGELOG (hunk 7 above)
Comment (waynebeaton) [reply CHANGELOG:13]: I don't understand what, specifically, you mean by "above".

File: eclipse_development_process.adoc (hunk 8 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:377]: It's a subtle point, but what's important is that all approval have been received for a _release_. So, strictly speaking, you're right... we don't need that for a progress review. However, we don't strictly need it for a release review either; we need it for a _release_.

File: history.adoc (hunk 10 above)
Comment (waynebeaton) [reply history.adoc:6]: I know, right?

File: eclipse_development_process.adoc (hunk 8 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:377]: So, for a progress review, we'd be looking for evidence that the IP Process is being followed., i.e. to "verify that the IP Policy has been followed<strike> and all approvals have been received</strike>?"

What's more important is that where we had previously been tying the IP Approvals to a review, we need to tie them to releases (which, as I stated in the previous comment, is where they should be). So, we need to include something in section 6.4, e.g. "The project lead must verify that the IP Policy has been followed and that all approvals have been received for all releases (major, minor, and service)."

Note that I have some definitions of major, minor, and service release in the "terms" contribution. I may try to roll some discussion around that into section 6.4 (it feels like an obvious opportunity to fill a hole).

File: eclipse_development_process.adoc (hunk 9 above)
Comment (waynebeaton) [reply eclipse_development_process.adoc:418]: I'm not sure what, specifically to add. IMHO, the statement is complete enough.

Comment (guw) [reply eclipse_development_process.adoc:418]: Ok.
